charmcraft/cmdbase.py

```diff
@@ -39,6 +41,7 @@ class BaseCommand(craft_cli.BaseCommand):
     """
 
     needs_config = False
+    needs_config_bases = False
```

Comment:
I understand that this implementation is what was agreed on, but to me it's awkward that we are adding "escape hatches" to bypass issues on the project file. Wouldn't it make more sense to not load the project at all, if the command doesn't need it?
My fear is this: what if tomorrow another restriction is added to charmcraft.yaml? We are going to add another `needs_config_...` to handle it.
@sergiusens can we discuss alternatives here? Or should we just do this quick-and-dirty?

Reply:
The things is current implementations all check configuration files before getting commands. If we change it to check the basic command first and then decide whether to load the corresponding configuration file, this problem can be eliminated.

Reply:
@tigarmo What do you mean "not load the project at all"?

The config needs to be parsed, the command needs to be chosen, and at that point decisions can be made. It's not super super cheap, but if you'll end up in an error, timings are not *that* important.

Regarding the how and when to take the decision, it can be done two ways I think:
- take the decision in the config structure using commands attributes (this is the one proposed here); it looks very organized, but may not scale well if different commands have different needs.
- take the decision in the command itself; for example, the first thing the pack command could do is `if self.config.bases is None: raise CraftError("Bases is needed for packing")`; it's not centralized in one place, but it's very flexible.